src/OrganisationRegistry.Api/Security/OrganisationRegistryTokenBuilder.cs

```diff
@@ -0,0 +1,122 @@
+
+        public string BuildJwt(ClaimsIdentity identity)
+        {
+            var plainTextSecurityKey = _configuration.JwtSharedSigningKey;
+            var signingKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(plainTextSecurityKey));
+            var signingCredentials = new SigningCredentials(signingKey, SecurityAlgorithms.HmacSha256);
+
+            var securityTokenDescriptor = new SecurityTokenDescriptor
+            {
+                Audience = _configuration.JwtAudience,
+                Issuer = _configuration.JwtIssuer,
+                Subject = identity,
+                SigningCredentials = signingCredentials,
+                IssuedAt = DateTime.UtcNow,
+                Expires = DateTimeOffset.Now.AddMinutes(_configuration.JwtExpiresInMinutes).UtcDateTime,
+                NotBefore = DateTime.UtcNow
+            };
+
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var plainToken = tokenHandler.CreateToken(securityTokenDescriptor);
+            var signedAndEncodedToken = tokenHandler.WriteToken(plainToken);
+
+            return signedAndEncodedToken;
+        }
+
+        public ClaimsIdentity ParseRoles(ClaimsIdentity identity)
+        {
+            identity.AddClaim(new Claim(OrganisationRegistryClaims.ClaimUserId, identity.GetClaim(JwtClaimTypes.Subject), ClaimValueTypes.String));
+            identity.AddClaim(new Claim(OrganisationRegistryClaims.ClaimName, JwtClaimTypes.FamilyName, ClaimValueTypes.String));
+            identity.AddClaim(new Claim(OrganisationRegistryClaims.ClaimFirstname, JwtClaimTypes.GivenName, ClaimValueTypes.String));
+
+            var roles = identity.GetClaims(OrganisationRegistryClaims.ClaimRoles)
+                .Select(x => x.ToLowerInvariant())
+                .Where(x => x.StartsWith(OrganisationRegistryClaims.OrganisationRegistryBeheerderPrefix))
+                .Select(x => x.Replace(OrganisationRegistryClaims.OrganisationRegistryBeheerderPrefix, ""))
+                .ToList();
+
+            var developers = _configuration.Developers?.Split(new[] {';'}, StringSplitOptions.RemoveEmptyEntries)
+                .Select(x => x.ToLowerInvariant());
+            if (developers != null &&
+                developers.Contains(identity.GetClaim(OrganisationRegistryClaims.ClaimAcmId).ToLowerInvariant()))
+                AddRoleClaim(identity, OrganisationRegistryApiClaims.Developer);
+
+            if (roles.Count <= 0)
+                return identity;
+
+            // OrganisationRegistryBeheerder-algemeenbeheerder,beheerder:OVO002949
+            // OrganisationRegistryBeheerder-beheerder:OVO001831
+            // OrganisationRegistryBeheerder-beheerder:OVO001834
+            // OrganisationRegistryBeheerder-orgaanbeheerder:OVO001834 // mag alle organen beheren, OVO nummer is puur technisch en nutteloos.
+
+            if (roles.Any(x => x.Contains(OrganisationRegistryClaims.OrganisationRegistryBeheerderRole)))
```

Comment:
het lijkt me wenselijk om deze geneste if eruit te refactoren ... dit soort constructies is nogal prone voor bugs ...